dom/events/scrolling/wheel-event-transactions-basic.tentative.html

```diff
@@ -0,0 +1,139 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+<meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1">
+<meta name="variant" content="?include=target-basic"/>
+<meta name="variant" content="?include=scroll-over-scrollable-child"/>
+<meta name="variant" content="?include=transaction-not-bound-to-scroll-frame"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="/common/subset-tests-by-key.js"></script>
+<script src="scroll_support.js"></script>
+<style>
+body {
+  margin: 0;
+  padding: 0;
+  height: 200vh;
+}
+
+.spacer {
+  width: 100%;
+  height: 25px;
+  padding: 0;
+  margin: 0;
+}
+
+#scrollableDiv {
+  width: 100%;
+  height: 500px;
+  overflow: scroll;
+  background: yellow;
+}
+
+#innerDiv {
+  width: 100%;
+  height: 4000px;
+  background: red;
+}
+
+</style>
+<head>
+<body onload=runTest()>
+<div id="firstRootSpacer" class="spacer" style="background: cyan"></div>
+<div id="secondRootSpacer" class="spacer" style="background: magenta"></div>
+<div id="scrollableDiv">
+  <div id="innerDiv">
+    <div id="firstInnerSpacer" class="spacer" style="background: green">
+    </div>
+    <div id="secondInnerSpacer" class="spacer" style="background: blue">
+    </div>
+  </div>
+</div>
+</body>
+
+<script>
+
+let variants = [
+  // Ensure that the wheel transaction fires all wheel events to the initial
+  // target.
+  {
+    key: 'target-basic',
+    origin: "viewport",
+    scrollX: 40,
+    scrollY: 2,
+    scrollingElement: document.scrollingElement,
+    targetElement: firstRootSpacer,
+    title: 'Wheel events should be captured to one target #1',
+  },
+  // Ensure that the wheel transaction fires all wheel events to the initial
+  // target, even when another scroll frame is under the mouse cursor.
+  {
+    key: 'scroll-over-scrollable-child',
+    origin: "viewport",
+    scrollX: 40,
+    scrollY: 27,
+    scrollingElement: document.scrollingElement,
+    targetElement: secondRootSpacer,
+    title: 'Wheel events should be captured to one target #2',
+  },
+  // Ensure that the wheel transaction targets the topmost-element, which is
+  // not the scrollable element.
+  {
+    key: 'transaction-not-bound-to-scroll-frame',
+    origin: innerDiv,
+    scrollX: 40,
+    scrollY: 2,
+    scrollingElement: scrollableDiv,
+    targetElement: innerDiv,
+    title: 'The wheel event transactions target may not be a scroll frame',
+  },
+];
+
+function runTest() {
+  async function testBasic(testInfo, t) {
+    await waitForCompositorReady();
+
+    await waitForCompositorCommit();
+
+    let wheelEventTargets = [];
+    function pushTargetToWheelEventTargets(e) {
+      wheelEventTargets.push(e.target)
```

Comment:
I wonder, how about to check `relatedTarget` too? They should be `secondRootSpacer` -> `firstInnerSpacer` -> `secondInnerSpacer` -> `innerDiv` -> `document.body` -> `document.documentElement`. The related target may be skipped by different scroll speed etc, but the order should never be changed. Therefore, store the target into an array and shift it if different `relatedTarget` is set, then, finally, the array becomes empty, make the test fail. This may pass if the related target is always same as the `eventTarget`, though (should check `.relatedTarget` never be same value?). So, up to you whether adding this test.

(If some browsers fail, I guess that it's tested isolated within `test()` in `testBasic` because the result is less important than `.target`.)

Reply:
Did a bit of research, and it would appear that no browser currently implements this. This is specified in [event-type-wheel](w3c.github.io/uievents/#event-type-wheel), but all tests I ran would result in `relatedTarget == null`. I think this would make a great follow-up issue for this project.